### ISSUE TITLE
LDAP auth doesn't support IPv6 addresses

### ISSUE THREAD

**Issue by Longsight** [metabase/metabase#12879]
**Describe the bug**
The LDAP host field does not support IPv6 addresses in any sensible format. Attempting to login with LDAP auth configured to use an IPv6 address fails with an LDAP connection error.

**Logs**
With an address of the form `2a00:1098:aaaa:bbbb::c`:
```
Jul 09 14:38:18 sahand java[19906]: 07-09 14:38:18 ERROR api.session :: Problem connecting to LDAP server, will fall back to local authentication
Jul 09 14:38:18 sahand java[19906]: LDAPException(resultCode=91 (connect error), errorMessage='An error occurred while attempting to resolve address '2a00':  UnknownHostException(2a00: Name or service not known), ldapSDKVersion=4.0.4, revision=27051')
```
With an address of the form `[2a00:1098:aaaa:bbbb::c]`:
```
Jul 09 14:58:20 sahand java[19906]: 07-09 14:58:20 ERROR api.session :: Problem connecting to LDAP server, will fall back to local authentication
Jul 09 14:58:20 sahand java[19906]: LDAPException(resultCode=91 (connect error), errorMessage='An error occurred while attempting to resolve address '[2a00':  UnknownHostException([2a00: invalid IPv6 address), ldapSDKVersion=4.0.4, revision=27051')
```

**To Reproduce**
1. Go to LDAP Admin
2. Enter the IPv6 address of a working LDAP host in the `LDAP HOST` field
3. Enter valid settings in other fields
4. Try to login using LDAP credentials

**Expected behavior**
Successful login.

**Actual results**
Failed login. Log examination indicates that connection to the LDAP server fails.

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10-LTS",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10-LTS",
    "os.name": "Linux",
    "os.version": "3.10.0-1127.10.1.el7.x86_64",
    "user.language": "en",
    "user.timezone": "Europe/London"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.8"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-05-28",
      "tag": "v0.35.4",
      "branch": "release-0.35.x",
      "hash": "b3080fa"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Blocking all use. We use IPv6 for all systems internally, and are deprecating IPv4. It's reasonable to assume that the LDAP SDK supports IPv6 addresses, and the errors suggest that the address is being clipped at the first `:` before being passed to the `LDAPConnection` instance.

**Comment by flamber** [comment]
Hi @Longsight 
A workaround is to create local alias in `/etc/hosts`, which you use in the Metabase interface (haven't tested, but should work).

**Comment by Longsight** [comment]
> Hi @Longsight
> A workaround is to create local alias in `/etc/hosts`, which you use in the Metabase interface (haven't tested, but should work).
I've done this for now for development purposes, but it's not ideal for deployment, as what I'm currently trying to do is deploy a number of instances with Puppet, and it's an extra moving part (that requires a very local alias, as the name must not resolve to a v4 address at all).
Thanks for looking at it in the meantime.

**Comment by camsaul** [comment]
Fixed by #12882
